### ISSUE TITLE
Fix map behavior

### ISSUE THREAD

**Issue by bakeiro** [Couchers-org/couchers#4847]
## Current situation
- We display all users of the platform, but once we search we return a very small subset 
- We don't have pagination
- These two things combined might confuse the user, since the behavior isn't consistent at all
- Unclear map behavior

### David's proporsal
- It's like trustroots, airbnb
- By default, shows the user's location zoomed out
- lLimit how much a user can zoom out
- Searchs automatically (or not) as the user moves the map
- Always returning a subset of the results
- Use pagination
- Show to the user "showing 100 out of 2000" to make them understand is a subset

### Jesse's proporsal
- Map shows all the pinpoints by default (since it's a big selling point)
- Pinpoints are ONLY re-render onces the user applies a filter, this returns a new subset for the whole world.
    - this means that applying a filter filters the whole world
    - that scales only if we generate cache for each filter combination in a static .json
    - we should regenerate that every night in a cronjob
    - data leak shouldn't be a problem since it only contains the coordinates and userId which makes sense only on our system
    - sounds overengineering honestly
- Left sidebar remains always empty (or hidden or whatever), until a user zooms enough in, afterwards a view box search (or whatever is called) is performed and then sidebar shows paginated results) map pinpoints remain unaltered
- When the user zooms out the sidebar gets empty again
- Clicking on a map's pinpoint: 
     - Ideally moves to the user's page and center it on the list, but we need to load all result's pages to do that
     - Alternative we just show 1 result on the list (the clicked pinpoint)
- All of this is to show a lot of users in the very first render (so user doesn't have to move around the map or zoom in to find this out)

**Comment by bakeiro** [comment]
@jesseallhands

**Comment by jesseallhands** [comment]
My suggestions for this:

Map shows zoomed-out (level 1) world view centered around home location of user. All pins are shown at all times until the user applies a filter.

Initially no list is shown on the left, but the searchbar is still usable (with a note below the searchbar to click on a map pin or apply a filter). 

If a user clicks on a "cluster" on the map, it zooms in to a closer level eg. level 1 to level 5, then if the user clicks on a single pin, it zooms to level 12 (centered around that pin) and shows the profile snippet on the left. If a user clicks on a cluster on level 5, then it zooms to level 12 (centered around the clicked cluster). This zooming behavior continues until the user clicks on a single pin or performs a search using filters (location/keyword/other). Once at level 12 or further zoomed in, clicking on a pin will no longer center the map around the pin.

Upon zooming into level 12 (eg. without clicking a pin), the paginated list appears on the left based upon pins shown on the screen. Pins that appear on the map are not affected by pagination (all pins that exist on the VISIBLE map are shown unless a filter is applied). However, the list on the left is paginated.

When a user searches a particular location, it zooms to level 12 centered on that location, and all pins on the visible map are shown (even if they fall outside of the area searched). For example, Madison is a relatively small city so it makes sense to show results in the nearby suburbs of Middleton and Fitchburg since they are also visible on the map.

When zooming out beyond level 12, the location filter is removed. If a keyword filter or other filter was applied, retain that filter unless the user clicks a "reset filters" button (currently there is no such button). When zooming out beyond level 12, all pins that appear on the visible map are shown, but the paginated list of results on the left disappears.

Basically the pins that appear on the map are in the paginated list based upon the visible map area, which is based upon the zoom level (1 being most zoomed out, 22 most zoomed in). This is visible in the map search URL, for example, 9.46 is the zoom level here: https://couchers.org/search#loc=9.46/40.8079/-73.9621

The behavior described above matches both trustroots and airbnb (though airbnb always has a paginated list on the left, which makes sense if you're a commercial platform/trying to sell something no matter what - but we are not a commercial platform).

**Comment by bakeiro** [comment]
As we discussed the 15/10/2024

> Some users don't care about the map, if the map fails/is slow we want to show only the list.

If that's the case 
the list shoudn't be separated from the map?

**Comment by bakeiro** [comment]
Jesse's proporsal summarized:

- Map shows all the pinpoints by default
- Pinpoints are ONLY re-render onces the user applies a filter, this returns a new subset for the whole world (we should cache that)
- Left sidebar remains always empty, until a user zooms enough in, afterwards a view box search (or whatever is called) is performed and then sidebar shows paginated results)
- When the user zooms out the sidebar gets empty again
- Clicking on a map's pinpoint: 
     - Ideally moves to the user's page and centered the list, but we need to load all pages results to do that
     - Alternative we just show 1 result on the list (the clicked pinpoint)

**Comment by bakeiro** [comment]
![11111111](https://github.com/user-attachments/assets/a28efa65-58c1-4aa0-bc35-eb2376dc89eb)

Personal opinnion: with my solution + showing the number users that subset has (not just the pagination results) and we update this as the user moves, users can notice we have a lot of users in out platform

**Comment by bakeiro** [comment]
an alternative I just though about:

- We show all users in the map as we do now
- Filters will filter out the users in the whole world
- Only once the user zooms in enough we perform a search (based on viewbox) and map pinpoints are filtered, we use pagination here
- Once the user moves out, we show all the users in the world again, showing the user we have a lot of accounts in our platform

basically it's a mix of both solutions...

**Comment by bakeiro** [comment]
this ticket was to made a decision, not to be worked by a dev, moving to blocked
